### PR TITLE
Adds threshold filter to the new home view

### PR DESCRIPTION
**Pull request by nzaytsev**
# Description

https://github.com/user-attachments/assets/b08cbd71-bc86-400f-9e03-ccc61f8f2ad7

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes include any required corresponding changes to the documentation (including CHANGELOG.md and README.md)
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

**Review by d13 — COMMENTED** [review]
Please be sure to separate the owner changes to another branch as we will not be shipping that this release.

File: src/webviews/apps/plus/home/components/overview.ts [hunk 1]
```diff
@@ -51,16 +55,32 @@ export class GlOverview extends SignalWatcher(LitElement) {
...
+					<gl-branch-threshold-filter
+						@gl-change=${(e: CustomEvent<{ threshold: OverviewRecentThreshold }>) => {
+							if (!this._overviewState.filter.stale || !this._overviewState.filter.recent) {
+								return;
+							}
+							this._ipc.sendCommand(SetOverviewFilter, {
+								stale: this._overviewState.filter.stale,
+								recent: { ...this._overviewState.filter.recent, threshold: e.detail.threshold },
+							});
+						}}
```
[the remark this answers is not in the document]
Comment (nzaytsev) [reply src/webviews/apps/plus/home/components/overview.ts:79]: done

File: src/webviews/apps/plus/home/components/overview.ts [hunk 2]
```diff
@@ -51,16 +55,32 @@ export class GlOverview extends SignalWatcher(LitElement) {
...
-
 		const { repository } = overview;
 		return html`
 			<div class="repository">
 				<gl-branch-section
 					label="Recent (${repository.branches.recent.length})"
 					.branches=${repository.branches.recent}
-				></gl-branch-section>
+				>
+					<gl-branch-threshold-filter
```
[the remark this answers is not in the document]
Comment (nzaytsev) [reply src/webviews/apps/plus/home/components/overview.ts:70]: done

File: src/webviews/apps/plus/home/components/branch-section.ts [hunk 3]
```diff
@@ -40,8 +43,7 @@ export class GlBranchSection extends LitElement {
 	override render() {
 		return html`
 			<div class="section">
-				<h3 class="section-heading">${this.label}</h3>
-				<slot></slot>
+				<h3 class="section-heading"><span>${this.label}</span><slot></slot></h3>
```
[the remark this answers is not in the document]
Comment (nzaytsev) [reply src/webviews/apps/plus/home/components/branch-section.ts:46]: done

File: src/webviews/apps/plus/home/components/branch-section.ts [hunk 4]
```diff
@@ -20,6 +20,9 @@ export const sectionHeadingStyles = css`
 		font-weight: normal;
 		margin-block: 0 0.8rem;
 		text-transform: uppercase;
+		display: flex;
+		justify-content: space-between;
+		gap: 8px;
```
[the remark this answers is not in the document]
Comment (nzaytsev) [reply src/webviews/apps/plus/home/components/branch-section.ts:25]: splitted

File: src/webviews/apps/plus/home/components/branch-threshold-filter.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+			case 'OneDay':
+				return '1 day';
+			case 'OneWeek':
+				return '1 week';
+			case 'OneMonth':
+				return '1 month';
+			case 'OneYear':
+				return '1 year';
+		}
+	}
```
[the remark this answers is not in the document]
Comment (nzaytsev) [reply src/webviews/apps/plus/home/components/branch-threshold-filter.ts:57]: done

File: src/webviews/apps/plus/home/components/branch-owners-filter.ts [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+import { GlElement } from '../../../shared/components/element';
+import '../../../shared/components/menu/index';
+import '../../../shared/components/menu/menu-item';
+import '../../../shared/components/menu/menu-list';
+import '../../../shared/components/overlays/popover';
+import { ipcContext } from '../../../shared/context';
+import type { HostIpc } from '../../../shared/ipc';
+
+@customElement('gl-branch-owner-filter')
+export class GlBranchOwnerFilter extends GlElement {
```
[the remark this answers is not in the document]
Comment (nzaytsev) [reply src/webviews/apps/plus/home/components/branch-owners-filter.ts:23]: done

**Review by d13 — APPROVED** [review]
There's some refinements we can make after merge, but LGTM.

**Pull request merged**